include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -0,0 +1,778 @@
+    configuration & operator=(configuration const &) = default;
+    configuration & operator=(configuration &&)      = default;
+    ~configuration()                                 = default;
+
+    //!\brief Constructs from std::tuple.
+    constexpr configuration(std::tuple<configs_t...> const & cfg) : base_type{cfg}
+    {}
+
+    //!\brief Constructs from std::tuple.
+    constexpr configuration(std::tuple<configs_t...> && cfg) : base_type{std::move(cfg)}
+    {}
+
+    /*!\brief Constructs a configuration from a variable of type seqan3::detail::configuration_fn_proxy or
+     *        seqan3::detail::configuration_fn_base
+     * \tparam cfg_fn_t Either seqan3::detail::configuration_fn_proxy or seqan3::detail::configuration_fn_base.
+     * \param cfg_fn    A variable of type `cfg_fn_t`.
+     *
+     * \details
+     *
+     * In some scenarios it might be possible, that only a single configuration element is used. In this case, the
+     * configuration must be constructed from a variable of type seqan3::detail::configuration_fn_proxy
+     * or seqan3::detail::configuration_fn_base. In order to construct a configuration object from it, this constructor
+     * invokes the functor with an empty configuration.
+     *
+     * ### Example
+     *
+     * ```cpp
+     * detail::configuration cfg = my_cfg(1, 2, 3);
+     * ```
+     * In the above example, we assume, that `my_cfg` is a variable of type seqan3::detail::configuration_fn_base.
+     */
+    template <typename cfg_fn_t>
+    //!\cond
+        requires is_type_specialisation_of_v<std::remove_reference_t<cfg_fn_t>, configuration_fn_proxy> ||
+                 std::is_base_of_v<configuration_fn_base<remove_cvref_t<cfg_fn_t>>, remove_cvref_t<cfg_fn_t>>
+    //!\endcond
+    constexpr configuration(cfg_fn_t && cfg_fn) :
+        configuration{std::invoke(std::forward<cfg_fn_t>(cfg_fn), configuration<>{})}
+    {}
+    //!\}
+
+    //!\cond
+    template <size_t elem_no, typename ... _configs_t>
+        requires (std::is_same_v<configs_t, _configs_t> && ...)
+    friend constexpr auto & seqan3::get(configuration<_configs_t...> & cfg) noexcept;
+
+    //!\copydoc seqan3::get()
+    template <size_t elem_no, typename ... _configs_t>
+        requires (std::is_same_v<configs_t, _configs_t> && ...)
+    friend constexpr auto const & seqan3::get(configuration<_configs_t...> const &) noexcept;
+
+    //!\copydoc seqan3::get<elem_no>(seqan3::detail::configuration)
+    template <size_t elem_no, typename ... _configs_t>
+        requires (std::is_same_v<configs_t, _configs_t> && ...)
+    friend constexpr auto && seqan3::get(configuration<_configs_t...> &&) noexcept;
+
+    //!\copydoc seqan3::get<elem_no>(seqan3::detail::configuration)
+    template <size_t elem_no, typename ... _configs_t>
+        requires (std::is_same_v<configs_t, _configs_t> && ...)
+    friend constexpr auto const && seqan3::get(configuration<_configs_t...> const &&) noexcept;
+    //!\endcond
```

Comment:
This block is unconditionally excluded from doxygen, but you still have copydocs inside :thinking: